### ISSUE TITLE
Bug: Declaration of type-only TSESTree namespace should be type-only syntactically

### ISSUE THREAD

**Issue by molisani** [typescript-eslint/typescript-eslint#11310]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

types

### Repro Code

```TypeScript
import { TSESTree } from "@typescript-eslint/types";
import type { TSESTree } from "@typescript-eslint/types";
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "verbatimModuleSyntax": true
  }
}
```

### Expected Result

The first line should have the following error, which can be solved by switching it to the second line.
```
'TSESTree' resolves to a type-only declaration and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.ts(1485)
```

### Actual Result

Both lines result in no errors. When this is transpiled with `verbatimModuleSyntax` the non-type-only `import` statement ends up in the transpiled .js output file which **fails at runtime** (as there is no TSESTree value import).

### Additional Info

This should be incredibly easy to fix, expect a PR shortly.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/types`             | `8.22.0` |
| `TypeScript`                           | `5.8.3` |

**Comment by molisani** [comment]
Apologies for not waiting for triage, but the corresponding fix is so short that I went ahead and created the PR in advance: https://github.com/typescript-eslint/typescript-eslint/pull/11311

**Comment by molisani** [comment]
In my haste for discovering a local fix to this issue I believe I have misdiagnosed the source of the error. I'll recreate this issue once I do some more research.
